test/puppetlabs/trapperkeeper/testutils/logging.clj

```diff
@@ -1,28 +1,321 @@
+  "Adds the specified appenders to the logger specified by logger-id,
+  calls f, and then removes them.  If logger-id is not a class, it
+  will be converted via str."
+  (let [logger (find-logger logger-id)]
+    (try
+      (doseq [appender appenders]
+        (.addAppender logger appender))
+      (f)
+      (finally
+        (doseq [appender appenders]
+          (.detachAppender logger appender))))))
+
+(defmacro with-additional-log-appenders
+  "Adds the specified appenders to the logger specified by logger-id,
+  evaluates body, and then removes them.  If logger-id is not a class,
+  it will be converted via str."
+  [logger-id appenders & body]
+  `(call-with-additional-log-appenders ~logger-id ~appenders (fn [] ~@body)))
+
+(defn call-with-log-appenders [logger-id appenders f]
+  "Replaces the appenders of the logger specified by logger-id with
+  the specified appenders, calls f, and then restores the original
+  appenders.  If logger-id is not a class, it will be converted via
+  str."
+  (let [logger (find-logger logger-id)
+        original-appenders (iterator-seq (.iteratorForAppenders logger))]
+    (try
+      (doseq [appender original-appenders]
+        (.detachAppender logger appender))
+      (call-with-additional-log-appenders logger-id appenders f)
+      (finally
+        (doseq [appender original-appenders]
+          (.addAppender logger appender))))))
+
+(defmacro with-log-appenders
+  "Replaces the appenders of the logger specified by logger-id with
+  the specified appenders, evaluates body, and then restores the
+  original appenders.  If logger-id is not a class, it will be
+  converted via str."
+  [logger-id appenders & body]
+  `(call-with-log-appenders ~logger-id ~appenders (fn [] ~@body)))
+
+(defn call-with-log-event-listener
+  "Calls (listen event) for each logger-id event produced during the
+  evaluation of f.  If logger-id is not a class, it will be converted
+  via str."
+  [logger-id listen f]
+  (with-started [listener (doto (log-event-listener listen) .start)]
+    (call-with-additional-log-appenders logger-id [listener] f)))
+
+(defmacro with-log-event-listener
+  "Calls (listen event) for each logger-id event produced during the
+  evaluation of body.  If logger-id is not a class, it will be
+  converted via str."
+  [logger-id listen & body]
+  `(call-with-log-event-listener ~logger-id ~listen (fn [] ~@body)))
+
+(defmacro with-logging-to-atom
+  "Conjoins all logger-id events produced during the evaluation of the
+  body onto the collection in the destination atom.  If logger-id is
+  not a class, it will be converted via str."
+  [logger-id destination & body]
+  `(with-log-event-listener
+     ~logger-id
+     (fn [event#] (swap! ~destination conj event#))
+     ~@body))
+
+(defn- suppressing-file-appender
+  [log-path]
+  (let [pattern "%-4relative [%thread] %-5level %logger{35} - %msg%n"
+        context (LoggerFactory/getILoggerFactory)]
+    (doto (FileAppender.)
```

Comment:
Don't think it's required per-se.  We could change it, though I suppose that could mean a lot more data in RAM given any heavy logging, and it'd leave no trace after a hard crash.